strings: prefix strings where it may be easily confused between noun/verb

lucasmz-dev opened the pull request:
makes it more reliable to reuse strings

lucasmz-dev commented:
@z-huang

I've been trying to get forks of InnerTune to recycle the work being done by contributors in our Weblate, potentially leading to contributions back as well, this worked with OuterTune and now I'm trying to get Malopieds' and Metrolist as well. What I realized (which was already in the back of my mind) is that a lot of strings may be confusing to know if it's a noun or verb, both for the translators and even code contributors, the Discord integration feature used the login string, which is actually meant as a noun, otherwise it would be "Log in". It also happened that "Sync" was used in OuterTune as a noun. When I analyzed how the translations differed and translated back with DuckDuckGo translate, it was quite clear to see that there were multiple instances where people disagreed, sometimes being wrong or right. I was also wrong on my initial translations IIRC.

As my job to organize people's work, I think this is pretty important to prevent quality issues. It isn't as obvious when there are issues with translations, so it's important to keep these as clear as possible.

I set up Weblate to mark translations coming from the repo as needing editing, so if this is merged, those strings (since they're technically new strings) will be marked for editing and translators will be notified specifically for these. I'll also add a warning to make sure people take note of the string name, and not just the string text.

This PR is just changing string names (and adding a few strings where they should exist) and pointing to them in the code.

If you can merge this, please do ASAP. I didn't wanna go the same route as the last Weblate PR because this is a bit of a more impactful thing, requiring forks that want to update stuff to also update this, I kept the number of strings to only what truly makes sense.

lucasmz-dev commented:
@mikooomich could you let me know if this looks alright to you?

mikooomich commented:
I like the idea, but introduces some problems while trying to fix the initial problem ex. If action_ and noun_ prefixes exist for some strings only, then wouldn't it be confusing given some strings are non-prefixed? 

>  I kept the number of strings to only what truly makes sense.

I understand why you did this but keep in mind everyone has their own interpretation of anything


Perhaps state the default translation for eveything is a noun, unless it has an action_ prefix? Not sure how to solve this tbh. Of course, I'd be open to changes

lucasmz-dev committed:
strings: prefix strings where it may be easily confused between noun/verb

lucasmz-dev committed:
translation: move translations to their new prefixed strings

weblate has been configured to mark new translations coming from the repo as needing editing, so translators will still have to take a 2nd look at this

lucasmz-dev commented:
Alright I see what you mean... I went and only prefixed the most confused strings with action_. This should be enough, the rest can probably be found through just context

TODO: On merge, Bulk edit: `key:=login` mark as editing

lucasmz-dev commented:
strings for login seem to be in the proper places
webview title uses 'login' and discord integration uses 'log in'
the rest should be ok as it's just replacing

lucasmz-dev commented:
was gonna merge with weblate but since we didn't get any conflicts, undoing. keeping weblate locked for now, however

lucasmz-dev commented:
thanks z!
